native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
+    import * as ReactNative from "react-native";
+    import { StatelessComponent } from "react";
+
+    interface StyledFunction<T> {
+      (values: TemplateStringsArray): T;
+      (...values: any[]): T;
```

Comment:
@DanielRosenwasser thanks! I've just changed that, I'll have a look at passing the types to the props functions, if you have any idea please shout! I'm still a newbie with TS :)